dbg.h

```diff
@@ -453,6 +457,16 @@ inline bool pretty_print(std::ostream& stream, const std::tuple<>&) {
   return true;
 }
 
+template <>
+inline bool pretty_print(std::ostream& stream, const time&) {
+  const auto ms =  std::chrono::duration_cast<std::chrono::microseconds>(std::chrono::high_resolution_clock::now().time_since_epoch()).count() % 1000000;
+  const auto hms = std::chrono::system_clock::to_time_t(std::chrono::system_clock::now());
```

Comment:
I think we should use the same timestamp here as for the `ms` computation. Otherwise, this could lead to inconsistencies.

I'll make the necessary changes.